libpromises/cf3lex.l

```diff
@@ -77,6 +79,9 @@ line       ^.*$
 
 comment    #[^\n]*
 
+macro_if    ^@if\ version_after\([0-9]{1,10}\.[0-9]{1,10}\)
```

Comment:
Does the user get a clear or confusing error if there is, e.g., two spaces after if. The rest of the language is very lenient about whitespace; perhaps this should also be. For example, @if\s+version_after\s*\(\s*[0-9]{1,10}\.[0-9]{1,10}\s*\). Forcing @if to start at the beginning of the line is too strict, in my opinion.